jobserver/models/user.py

```diff
@@ -47,11 +47,17 @@ def order_by_name(self):
 class UserManager(DjangoUserManager.from_queryset(UserQuerySet)):
     use_in_migrations = True
 
-    def create_superuser(self, email, password, **extra_fields):
-        username = email
-        return super().create_superuser(
-            username, email, password, roles=[CoreDeveloper]
-        )
+    def create_user(self, username, email=None, password=None, **extra_fields):
+        # We think we want the special handling from _create_user around the
+        # username, email, and password fields. But our User model extends
+        # AbstractBaseUser and doesn't have is_staff or is_superuser as
+        # django.contrib.auth.models.UserManager expects. So we call the #
+        # private function, which is not ideal as it breaks intended
+        # encapsulation. At least it will break loudly if a package update
+        # removes this.
+        # TODO: Consider extending BaseUserManager instead. Or possibly we
+        # don't need custom behavior around our User model at all. Ref #4625.
```

Comment:
I will create an issue. I'd like a pointer in the code for linkage to help future-dev realize that exists and avoid going through the same thought process and investigation (and possibly even issue-created if they don't find the existing one).

Mixed feelings on TODOs without issues. Most of them should have issues instead or just be fixed (speaking generally, not our specific TODOs). There might be a place for noting "this could use refactoring/change" but where it's neither trivial nor worth doing except as part of some larger work.

I see it's already in the cabinet for discussion tomorrow :-)

Reply:
#4627